test/e2e/suite/command/custom_header_test.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package command
+
+import (
+	"fmt"
+	"strings"
+
+	. "github.com/onsi/ginkgo/v2"
+	"oras.land/oras/test/e2e/internal/testdata/foobar"
+	. "oras.land/oras/test/e2e/internal/utils"
+)
+
+var _ = Describe("Common registry users:", func() {
+	When("custom header is provided", func() {
+		It("attach", func() {
+			testRepo := attachTestRepo("simple")
+			tempDir := CopyTestDataToTemp()
+			subjectRef := Reference(Host, testRepo, foobar.Tag)
+			prepare(Reference(Host, ImageRepo, foobar.Tag), subjectRef)
+			ORAS("attach", "--artifact-type", "test.attach", subjectRef,
+				fmt.Sprintf("%s:%s", foobar.AttachFileName, foobar.AttachFileMedia),
+				"-d", "-H", FoobarHeaderInput, "-H", AbHeaderInput).
+				WithWorkDir(tempDir).MatchRequestKeywords("Request headers:", FoobarHeader, AbHeader).Exec()
+		})
+		It("blob", func() {
+			blobDigest := "sha256:2c26b46b68ffc68ff99b453c1d30413413422d706483bfa0f98a5e886266e7ae"
+			ORAS("blob", "fetch", Reference(Host, ImageRepo, blobDigest), "--descriptor",
+				"-d", "-H", FoobarHeaderInput, "-H", AbHeaderInput).
+				MatchRequestKeywords("Request headers:", FoobarHeader, AbHeader).Exec()
+		})
+		It("manifest", func() {
+			manifest := `{"schemaVersion":2,"mediaType":"application/vnd.oci.image.manifest.v1+json","config":{"mediaType":"application/vnd.oci.image.config.v1+json","digest":"sha256:fe9dbc99451d0517d65e048c309f0b5afb2cc513b7a3d456b6cc29fe641386c5","size":53},"layers":[]}`
```

Comment:
Better to use `manifest fetch` to avoid hard-coding manifest.

Reply:
Replaced with manifest fetch.